crates/hir/src/semantics/source_to_def.rs

```diff
@@ -163,7 +164,10 @@ impl SourceToDefCtx<'_, '_> {
                             .include_macro_invoc(crate_id)
                             .iter()
                             .filter(|&&(_, file_id)| file_id == file)
-                            .flat_map(|(call, _)| {
+                            .flat_map(|(call, file_id)| {
+                                self.cache
+                                    .included_file_cache
+                                    .insert(*file_id, MacroFileId { macro_call_id: *call });
```

Comment:
This needs to happen somewhere else I think (its luck that the tests pass as is right now)